src/lightgbm/src/main/scala/LightGBMClassifier.scala

```diff
@@ -73,7 +73,7 @@ class LightGBMClassificationModel(
   featuresColName: String, predictionColName: String, probColName: String,
   rawPredictionColName: String, thresholdValues: Option[Array[Double]])
     extends ProbabilisticClassificationModel[Vector, LightGBMClassificationModel]
-    with ConstructorWritable[LightGBMClassificationModel] {
+    with ConstructorWritable[LightGBMClassificationModel] with DefaultParamsWritable {
```

Comment:
This should either use one or the other. If it uses Constructor Writable it should be parameterized by the constructor alone, if it uses DefaultParamsWritable it should have an empty constructor and use SparkML's Parameter class to pass the state on to the model. Please let me know if you need me to explain in more depth!